preprocessing/BETDisk/BETTranslatorInterface.py

```diff
@@ -0,0 +1,844 @@
+def readXROTORFile(xrotorFileName):
+    """
+    This functions read in the Xrotor filename provided.
+    it does rudimentary checks to make sure the file is truly in the Xrotor format.
+    :param xrotorFileName: string
+    :return: a dictionary with all the required values. That dictionary will be used to create BETdisks section of the
+            Flow360 input JSON file.
+
+
+    xrotor Input file has the following definitions:
+    Case run definition:
+    rho :air density (dimensional: kg/m3)
+    vso aka cinf : speed of sound ( dimensional: m/s)
+        !   RMU         Fluid dynamic viscosity  (dimensioned)
+        !   VSO         Fluid speed of sound     (dimensioned)
+        !   VEL         Flight speed             (dimensioned)
+        !   RAD         Rotor tip radius         (dimensioned)
+    rmu: Fluid dynamic viscosity  (dimensional (kg/ms) standard air:1.789E-05)
+    Alt: Altitude for fluid properties (km),  999.0 if not defined
+    Rad: rotor Tip radius dimensional (m)
+    Vel: flight speed dimensional (m/s)
+    Adv: Advance ratio (Vel/Vtip) where Vtip = propeller tip speed
+    Rake: unused- Winglet/droop type tips. We assume a planar propeller.
+    xi0:Blade root radial coordinate value (dimensional (m))
+    xiw: hub wake displacement radius (unused)
+    nAeroSections aka naero: number of AERO sections the blade is defined by, NOT TO BE CONFUSED WITH nGeomStations (AKA II) WHICH DEFINE THE BLADE GEOMETRY
+    xrotorInputDict stores all the blade sectional information as lists of nsection elements
+    rRsection: r/R location of this blade section
+    Aerodynamic definition of the blade section at xiaero
+        a0deg: angle of zero lift in degrees
+        dclda: Incompressible 2-d lift curve slope in radians
+        clmax: Max cl after which we use the post stall dc/dalfa (aka dclda_stall)
+        clmin: Min cl before which we use the negative alpha stall dc/dalfa (aka dclda_stall)
+
+        dclda_stall: 2-d lift curve slope at stall
+        dcl_stall: cl increment, onset to full stall
+        cmconst: constant Incompressible 2-d pitching moment
+        mcrit: critical Mach #
+        cdmin: Minimum drag coefficient value
+        cldmin: Lift at minimum drag coefficient value
+        dcddcl2: Parabolic drag param d(Cd)/dcl^2
+        reyref: reference Reynold's number
+        reyexp: Reynold's number exponent Cd~Re^rexp
+
+    nGeomStations: number of geometric stations where the blade geometry is defined at
+    nBlades: number of blades on the propeller
+    Each geometry station will have the following parameters:
+      r/R: station r/R
+      c/R: local chord divided by radius
+      beta0deg: Twist relative to disk plane. ie symmetric 2D section at beta0Deg would create 0 thrust, more beta0deg means more local angle of attack for the blade
+      Ubody: (unused) Nacelle perturbation axial  velocity
+
+    """
+    fid = open(xrotorFileName, 'r')
```

Comment:
there should be corresponding `close()` for this file. Either use `with` or:
```python
try:
    fid = open(xrotorFileName, 'r')
finally:
    fid.close()
```

we use `try ... finally ...` as the code might raise exception and we want to always close the file